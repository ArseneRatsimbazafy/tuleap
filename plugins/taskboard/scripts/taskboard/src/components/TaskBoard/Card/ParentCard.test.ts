/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount } from "@vue/test-utils";
import ParentCard from "./ParentCard.vue";
import { createStoreMock } from "@tuleap-vue-components/store-wrapper-jest";

jest.useFakeTimers();

describe("ParentCard", () => {
    it("displays a parent card", () => {
        const wrapper = shallowMount(ParentCard, {
            mocks: {
                $store: createStoreMock({ state: { user: { user_has_accessibility_mode: false } } })
            },
            propsData: {
                card: {
                    id: 43,
                    color: "lake-placid-blue",
                    background_color: "fiesta-red"
                }
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    it("doesn't add a dummy taskboard-card-background- class if the card has no background color", () => {
        const wrapper = shallowMount(ParentCard, {
            mocks: {
                $store: createStoreMock({ state: { user: { user_has_accessibility_mode: false } } })
            },
            propsData: {
                card: {
                    id: 43,
                    color: "lake-placid-blue",
                    background_color: ""
                }
            }
        });
        expect(wrapper.classes()).not.toContain("taskboard-card-background-");
    });

    it("adds accessibility class if user needs it", () => {
        const wrapper = shallowMount(ParentCard, {
            mocks: {
                $store: createStoreMock({ state: { user: { user_has_accessibility_mode: true } } })
            },
            propsData: {
                card: {
                    id: 43,
                    color: "lake-placid-blue",
                    background_color: ""
                }
            }
        });
        expect(wrapper.contains(".taskboard-card-accessibility")).toBe(true);
        expect(wrapper.classes()).toContain("taskboard-card-with-accessibility");
    });

    it("removes the show classes after 500ms", () => {
        const wrapper = shallowMount(ParentCard, {
            mocks: {
                $store: createStoreMock({ state: { user: { user_has_accessibility_mode: false } } })
            },
            propsData: {
                card: {
                    id: 43,
                    color: "lake-placid-blue",
                    background_color: "fiesta-red"
                }
            }
        });
        jest.runAllTimers();
        expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), 500);
        expect(wrapper.classes()).not.toContain("taskboard-card-show");
    });
});
