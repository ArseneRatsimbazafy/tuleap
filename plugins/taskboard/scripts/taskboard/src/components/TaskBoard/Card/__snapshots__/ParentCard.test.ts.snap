// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParentCard displays a parent card 1`] = `
<div
  class="taskboard-card taskboard-card-parent taskboard-card-lake-placid-blue taskboard-card-show"
>
  <div
    class="taskboard-card-content"
  >
    <card-xref-label-stub
      card="[object Object]"
    />
  </div>
</div>
`;
